### PR TITLE
Endpoints ADR

### PR DESCRIPTION
**Pull request by andreasphil**

### CODE REVIEW

File: doc/adr/0000-template.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# 0. Template
+
+Date: 2024-03-07
```
Comment (hannes-mk) [inline doc/adr/0000-template.md]: Maybe have the date as `20xx-xx-xx`?

File: doc/adr/0000-template.md
Comment (hannes-mk) [inline doc/adr/0000-template.md:1]: Having this as a non-tooling based template was a great idea, btw. :pray:

File: doc/adr/0007-api-endpoints-setup.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# 7. API endpoints setup
+
+Date: 2024-03-07
```
Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md:3]: Just curious: Was that the date where we actually agreed?

File: doc/adr/0007-api-endpoints-setup.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## Context
+
+We need to support a growing list of endpoints. At the same time, we want to avoid repeated discussions about what each endpoint should look like, and reduce the time spent thinking about which information should be available where.
+
+This ADR outlines our idea of sensible defaults for how endpoints should be structured, and how they should behave. The goal is to have a shared understanding that makes API endpoint-related decisions easier, and the structure and behavior of those endpoints more predictable.
+
+## Decision
+
+Our proposal consists of a simple foundation that, while not ideal, supports everything we need. It is also easy to extend to support requirements we will very likely have in the near future.
```
Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md]: I'd scrape the "while not ideal" and just go for "that fulfills our current needs."

Comment (hannes-mk) [reply doc/adr/0007-api-endpoints-setup.md]: Maybe "... current needs and is also...".

File: doc/adr/0007-api-endpoints-setup.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+We're generally not dogmatic, but we're aiming for a REST-ful interface, meaning that we ask the API for resources, not functionality.
```
Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md]: Maybe the fact first: "We're aiming for a RESTful interface. meaning ... - but we're not dogmatic about it."

File: doc/adr/0007-api-endpoints-setup.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+### Foundation
+
+We provide create, read, update, and delete endpoints for individual resources (not all resources need to support all operations), as well as list endpoints for listing all available resources.
```
Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md]: Not sure if the last part could be misunderstood: Would
"... all available resources of a certain type." be more precise?

File: doc/adr/0007-api-endpoints-setup.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+};
+```
+
+...would correspond to the following endpoints:
+
+| Operation                      | Notes                                                        |
+| ------------------------------ | ------------------------------------------------------------ |
+| `GET /laws`                    | Returns an array of `{ eli: string; title: string }`         |
+| `GET /laws/:eli`               | Returns a single law of type `{ eli: string, title: string}` |
+| `GET /laws/:eli/articles`      | Returns the articles inside the law                          |
```
Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md]: maybe "...articles of the law" instead of "...articles in the law"?

Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md]: Do you want to illustrate the JSON for the last two endpoints as well?

File: doc/adr/0007-api-endpoints-setup.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+| `GET /laws/:eli/articles/:eid` | Returns a single article                                     |
+
+### Extension 1: Support for nested resources
+
+In the above example, having to know that laws contain articles as well as knowing how to fetch them is inconvenient, and often also inefficient. We're planning to solve this eventually by including references to nested resources inside the parent resource following the [Hypertext Application Language](https://en.wikipedia.org/wiki/Hypertext_Application_Language) convention.
+
+### Extension 2: Supporting different data formats
```
Comment (hannes-mk) [inline doc/adr/0007-api-endpoints-setup.md:60]: This is already the case. Should this be reflected in the document?

**Review by hannes-mk — APPROVED** [review]

File: doc/adr/0007-api-endpoints-setup.md (hunk 2 above)
Comment (andreasphil) [reply doc/adr/0007-api-endpoints-setup.md:3]: It was the date where the ADR was written because I forgot when we argeed :D

**Pull request merged**